daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -803,6 +805,74 @@ case class FNFloor(recipe: CompiledDPath, argType: NodeInfo.Kind) extends FNOneA
   }
 }
 
+/**
+ * This is "logical" shift left, meaning it sign-extends a negative signed value.
+ * dfdl:leftShift for signed values does sign-extension, for logical (zero-filling) left shift you must use an unsigned type
```

Comment:
Comment says `dfdl:leftShift` but should be `dfdlx:leftShift`, with the `x`. 

Also, I think Mike mentioned a comment about this, but maybe it was some where else? Sign extending is only for right shift? There is no sign extending for left shift.